libs/hiraya/src/components/Button/index.tsx

```diff
@@ -1,12 +1,24 @@
-import { Button } from '@mui/material';
+import { ButtonProps, ButtonStyled } from './style';
+import { buttonIconOnly } from './theme';
 
-interface ButtonProps {
-  label: string;
-  variant?: 'text' | 'outlined' | 'contained';
-}
+const Button: React.FC<ButtonProps> = ({ label, iconOnly, ...props }) => {
+  const renderContent = () => {
+    if (!iconOnly) {
+      return label;
+    }
+    return buttonIconOnly();
+  };
 
-const ButtonUsage: React.FC<ButtonProps> = ({ ...props }) => {
-  return <Button variant={props.variant}>{props.label}</Button>;
+  return (
+    <ButtonStyled
+      variants={props.variants}
+      buttonHeightSizes={props.buttonHeightSizes}
+      isFullWidth={props.isFullWidth}
+      iconOnly={iconOnly}
```

Comment:
Let's remove all of these except label and iconOnly and pass the rest via {...props}

Reply:
That way we have less debt in the code. The lesser you write code, the lesser code you have to maintain, that would be our mantra.